src/pkg/cli/compose/validation.go

```diff
@@ -43,17 +43,24 @@ func ValidateProject(project *composeTypes.Project) error {
 		return services[i].Name < services[j].Name
 	})
 
-	errs := make([]error, len(services))
+	var errs []error
+	for _, svccfg := range services {
+		errs = append(errs, validateService(&svccfg, project))
+	}
 	for i, svccfg := range services {
-		errs[i] = validateService(&svccfg, project)
+		for j := i + 1; j < len(services); j++ {
+			if svccfg.Name == services[j].Name {
+				errs = append(errs, fmt.Errorf("service %q defined multiple times", svccfg.Name))
```

Comment:
Do we need this error? This case will always trigger the next error too. Or, we add a `continue` here to avoid two errors for the same service.
```suggestion
				errs = append(errs, fmt.Errorf("service %q defined multiple times", svccfg.Name))
				continue
```